contracts/lifecycle/TimeFrame.sol

```diff
@@ -0,0 +1,70 @@
+pragma solidity ^0.5.0;
+
+/**
+ * @title Provides generic functionality for a TimeFrame.
+ * @author Sam Ruberti <sam@GRAFFT.io>
+ * @dev This library provides useful methods for governing a start and an end time.
+ *
+ * WARNING: This should NOT be used for contracts that depend on time for an outcome where a miner can benefit.
+ * One example would be a lottery where the first submission of an answer to a problem is rewarded. In this case the
+ * contract would be vulnerable to malicious miners who could read the answer of a not-yet-mined transaction then
+ * substitute their own answer in a new transaction before the victim's transaction, stealing the reward in the process.
+ * The miner could steal the reward.
+ *
+ * SAFE USE: TimeFrame CAN be used for approximate times that govern periods when methods and actions may occur.
+ * An example is an escrow that contains a start and expiration date. You can read about timestamp security
+ * vulnerabilities here:
+ * github.com/ethereumbook/ethereumbook/blob/develop/09smart-contracts-security.asciidoc#block-timestamp-manipulation
+ */
+
+
+/* solhint-disable not-rely-on-time, security/no-block-members */
+library TimeFrame {
+    struct Epoch {
+        uint256 start;
```

Comment:
To emphasize the fact that these values should not be accessed directly (i.e. all interactions with an `Epoch` should go through the library), we could use leading underscores here. I did the same for `Counters` [here](https://github.com/OpenZeppelin/openzeppelin-solidity/pull/1610/files#diff-c07b8585b0773a167b8550176eed16abR23).